test/suites/integration/security_group_test.go

```diff
@@ -53,6 +56,38 @@ var _ = Describe("Subnets", func() {
 
 		env.ExpectInstance(pod.Spec.NodeName).To(HaveField("SecurityGroups", ContainElement(&securityGroups[0])))
 	})
+
+	It("should have an update AWSNodeTemplate stauts for security groups", func() {
+		subnets := getSubnets(map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName})
+		Expect(len(subnets)).ToNot(Equal(0))
+		shuffledAZs := lo.Shuffle(lo.Keys(subnets))
+
+		provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+			AWS: v1alpha1.AWS{
+				SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+				SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			},
+		})
+		provisioner := test.Provisioner(test.ProvisionerOptions{
+			ProviderRef: &v1alpha5.ProviderRef{Name: provider.Name},
+			Requirements: []v1.NodeSelectorRequirement{
+				{
+					Key:      v1.LabelZoneFailureDomainStable,
+					Operator: "In",
+					Values:   []string{shuffledAZs[0]},
+				},
+			},
+		})
+		pod := test.Pod()
+
+		env.ExpectCreated(pod, provider, provisioner)
+		env.EventuallyExpectHealthy(pod)
+		env.ExpectCreatedNodeCount("==", 1)
+
+		var ant v1alpha1.AWSNodeTemplate
+		Expect(env.Client.Get(env, client.ObjectKeyFromObject(provider), &ant)).To(Succeed())
+		Expect(len(ant.Status.SecurityGroupIDs)).ToNot(BeZero())
```

Comment:
This is a little different than what I was expecting to happen. Looks like on line 73 we're using node selector requirements on the provisioner to choose a specific AZ, but I thought since this is the AWSNodeTemplate's Status that we shouldn't be concerned about what the provisioner is. We only watch AWSNodeTemplate events too in the controller, so I think this will always succeed.